src/puppetlabs/puppetdb/command.clj

```diff
@@ -363,12 +364,15 @@
     (log-command-processed-messsage id received start-time :deactivate-node certname)))
 
 (defn deactivate-node [{:keys [payload version] :as command} start-time db]
-  (-> command
-      (assoc :payload (case version
-                        1 (deactivate-node-wire-v1->wire-3 payload)
-                        2 (deactivate-node-wire-v2->wire-3 payload)
-                        payload))
-      (deactivate-node* start-time db)))
+  (let [validated-payload (upon-error-throw-fatality
```

Comment:
This makes the `upon-error-throw-fatality` check in `deactivate-node-wire-v1->wire-3` redundant, so that can probably get removed

Reply:
updated